### ISSUE TITLE
Put operators back onto Observable

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/horizon#532]
At the moment, the default build of the client contains no operators on Observable. This is great for producing a smaller library, but it comes at the cost of making it annoying to add them back. Internally, we use the Stage-0 [function bind syntax](https://github.com/zenparsing/es-function-bind) `::` which babel provides, to apply operators without having to patch them onto the Observable and export them. In practice though, this doesn't save us much space, since the code for those operators still needs to be compiled into the client.

I think we should patch in the operators we use (share, map, merge etc) and a few more useful ones like `defaultIfEmpty`. We can add new ones in the future as needed.

**Comment by deontologician** [comment]
An added benefit is that this eases the transition to typescript, since typescript doesn't support `::`

**Comment by cschram** [comment]
It looks like this should be as simple as changing import statements from this:

```
import { filter } from 'rxjs/operator/filter'
```

to this:

```
require('rxjs/add/operator/filter')
```

And then changing call sites to use `.` instead of `::`.

**Comment by niieani** [comment]
@cschram You don't need to change an `import` to a `require`, you can just as easily `import` for side-effects only, like this:

``` ts
import 'rxjs/add/operator/filter'
```

**Comment by deontologician** [comment]
Fixed by @cschram in 36bf737
